includes/core/README.md

```diff
@@ -1,3 +1,105 @@
+
+## Available functionality
+
+WooCommerce Payments Core is under active development, and currently includes the APIs, listed below.
+
+### Gateway Mode
+
+This is a singular `WCPay\Core\Mode` object, accessible through `WC_Payments::mode()`. It allows you to check whether the gateway is in any of the listed modes, as well as to force it into one of them:
+
+| Mode     | Check      | Setter   | Description                           |
+|----------|------------|----------|---------------------------------------|
+| __Live__ | `is_live()`| `live()` | Standard working mode of the gateway. |
+| __Test__ | `is_test()`| `test()` | Allows test mode on live accounts.    |
+| __Dev__  | `is_dev()` | `dev()`  | Allows test accounts to be created.   |
+
+During initialization, the following logic is used:
+
+1. Dev mode would be entered if:
+    - Either [WordPress's environment type](https://developer.wordpress.org/reference/functions/wp_get_environment_type/#description) is either `development` or 		`staging`.
+    - or `WCPAY_DEV_MODE` is defined and set to boolean true.
+2. Test mode is entered if:
+    - Either Dev mode is already enabled.
+    - or the gateway's test mode setting is on.
+3. If the gateway is neither in dev or test mode, live mode is entered.
+
+To alter this behavior, you can use the `wcpay_dev_mode` and `wcpay_test_mode` filters, for example:
+
+```php
+add_filter( 'wcpay_dev_mode', '__return_true' );
+```
+
+This is done on the `init` action.
+
+> 🎗️ The object is not accessible before the `init` WordPress action.
+
+Once the mode has been initialized, the gateway will work in that mode. The setters (from the table above) may be used to switch the working mode during runtime, but it should be done with caution, as other functionality may already be initialized in the main mode, hence it is recommended to switch back once done.
+
+__Checking the current mode:__
+
+```php
+// Check for live mode:
+WC_Payments::mode()->is_live();
+
+// Check for test mode:
+WC_Payments::mode()->is_test();
+
+// Check for dev mode:
+WC_Payments::mode()->is_dev();
+```
+
+### Services
+
+WooCommerce Payments contains a comprehensive set of internal services, but as explained in the beginnig of this document, they are subject to change without notice. Instead, please use the services, available in core:
+
+- [🔗 `WC_Payments_Customer_Service_API`](services/customer-service.md)
+
+### Server requests & responses
+
+Communication with the WooCommerce Payments Server is currently done through a mix of the `WC_Payments_API_Client`, and request classes, and being migrated to fully use the latter.
```

Comment:
I prefer people not looking there, LOL! 😆 

Joke aside, I'm intentionally trying not to link files in order to avoid broken links in the future. Linking to other markdown files is reliable, as the links are relative to the current doc. Linking to code would require a change in the link if anything changes (switch away from GH, change repo name or owner, change main branch, rename the file, or something else). I'm mostly concerned about renamed files, as soon we'd like to add autoloading, and it will definitely require batch changing of a lot of file names.